extension/healthcheckextension/healthcheckextension_test.go

```diff
@@ -71,6 +72,45 @@ func TestHealthCheckExtensionUsageWithoutCheckCollectorPipeline(t *testing.T) {
 	require.Equal(t, http.StatusServiceUnavailable, resp2.StatusCode)
 }
 
+func TestHealthCheckExtensionUsageWithCustomizedPathWithoutCheckCollectorPipeline(t *testing.T) {
+	config := Config{
+		TCPAddr: confignet.TCPAddr{
+			Endpoint: testutil.GetAvailableLocalAddress(t),
+		},
+		CheckCollectorPipeline: defaultCheckCollectorPipelineSettings(),
+		Path:                   "/health",
+	}
+
+	hcExt := newServer(config, zap.NewNop())
+	require.NotNil(t, hcExt)
+
+	require.NoError(t, hcExt.Start(context.Background(), componenttest.NewNopHost()))
+	t.Cleanup(func() { require.NoError(t, hcExt.Shutdown(context.Background())) })
+
+	// Give a chance for the server goroutine to run.
+	runtime.Gosched()
```

Comment:
This smells like a new flaky test :-) If httptest doesn't help with the lifecycle, can you use a "[require.Eventually](https://pkg.go.dev/github.com/stretchr/testify/require#Eventually)" instead?

Reply:
Have replaced it with `require.Eventually`. But I only touched the two tests I added, not sure if I should also update the existing ones?

Reply:
Touching only yours is good enough already. Thanks!